packages/dds/tree/src/feature-libraries/default-schema/mappedEditBuilder.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { FieldUpPath, UpPath } from "../../core/index.js";
+import type {
+	IDefaultEditBuilder,
+	OptionalFieldEditBuilder,
+	SequenceFieldEditBuilder,
+	ValueFieldEditBuilder,
+} from "./defaultEditBuilder.js";
+
+/**
+ * An IDefaultEditBuilder implementation based on another IDefaultEditBuilder that uses a different content type for insertions.
+ */
+export class MappedEditBuilder<TBase, TAdapted> implements IDefaultEditBuilder<TAdapted> {
```

Comment:
Some context on why this file is added in this PR would be useful.

Reply:
Added the following to the PR description:

`MappedEditBuilder` was added to make it easy for `LazyField` editing methods to keep taking in insertion content as cursors while the underlying editors use `TreeChunk`s.